### PR TITLE
use requirejs map to achieve private jq

### LINKED ISSUE #1990: Bokeh should use a private jquery
We still aren't getting this right.

Reference: https://github.com/bokeh/rbokeh/issues/25#issuecomment-76252549

### PR DESCRIPTION
**Pull request by bryevdv**
issues: fixes #1990

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap @tswicegood please comment. This appears to work, if I load a Bokeh plot, then `window.$` is undefined (it was not before, it had the version loaded in our `config.coffee`) Only question: in `jquery-private.coffee` do we also need to try to save/restore `window.$`?

File: bokehjs/src/coffee/main.coffee [hunk 1]
```diff
@@ -18,17 +18,9 @@ define (require, exports, module) ->
 
   # binding the libs that bokeh uses so others can reference them
   Bokeh._                 = require("underscore")
-  Bokeh.$                 = require("jquery")
+  Bokeh.$                 = require("jquery").noConflict(true)
```
Comment (bryevdv) [inline bokehjs/src/coffee/main.coffee]: I think this can just be `require("jquery")` now

**Comment by bryevdv** [comment]
@mattpap I added `jquery` to alot to modules that were using `$` globally (apparently) but test still failing. I think it has to do with the jq ui shim. Any ideas?

**Comment by bryevdv** [comment]
For reference this is what I used to locate offending files:
```
find . -name "*.coffee" -exec grep -lR '[^\.@]\$' {} \; | xargs grep -L 'jquery'
```

**Comment by tswicegood** [comment]
@bryevdv my bet is that jquery-ui is failing because it's trying to load the global `window.jQuery` and modify that.  This change appears to have pulled it only into the scope of the files its loaded on so I believe you have to update the dependencies so it finds it correctly.  Not sure on that, but that's the hunch I'd start trying to track down if I were debugging it.

As for the actual change, this all looks reasonable.

**Comment by bryevdv** [comment]
Yah that was my guess too I was not able to find if there is some way in the shim config to make it see  the private `$` or if we need to actually edit the vendor source somehow.

**Comment by mattpap** [comment]
@bryevdv, it's not jquery-ui that's failing (at least for me), but jquery-event-*, because those aren't AMD modules. That was fixed in node.js PR.

**Comment by bryevdv** [comment]
Fantastic, JS tests pass now

**Comment by mattpap** [comment]
I wanted to ask where this `map` thing came from, but I saw this http://requirejs.org/docs/jquery.html. I wonder why I didn't notice this before. Anyway, LGTM.

**Comment by damianavila** [comment]
>I wanted to ask where this map thing came from, but I saw this http://requirejs.org/docs/jquery.html. I wonder why I didn't notice this before. Anyway, LGTM.

Interesting...

Also... LGTM

**Pull request merged**